windows/devices/uwb/UwbDeviceConnector.cxx

```diff
@@ -395,46 +395,45 @@ UwbDeviceConnector::SetApplicationConfigurationParameters(uint32_t sessionId, st
-    //
-
-    // auto& setParamsBuffer = applicationConfigurationParameters.DdiBuffer();
-    // auto& setParams = applicationConfigurationParameters.DdiParameters();
-
-    // // Allocate memory for the UWB_SET_APP_CONFIG_PARAMS_STATUS structure, and pass this to the driver request
-    // auto statusSize = offsetof(UWB_SET_APP_CONFIG_PARAMS_STATUS, appConfigParamsStatus[setParams.appConfigParamsCount]);
-    // std::vector<uint8_t> statusBuffer(statusSize);
-    // BOOL ioResult = DeviceIoControl(handleDriver.get(), IOCTL_UWB_SET_APP_CONFIG_PARAMS, std::data(setParamsBuffer), std::size(setParamsBuffer), std::data(statusBuffer), statusSize, nullptr, nullptr);
-    // if (!LOG_IF_WIN32_BOOL_FALSE(ioResult)) {
-    //     HRESULT hr = HRESULT_FROM_WIN32(GetLastError());
-    //     PLOG_ERROR << "error when sending IOCTL_UWB_SET_APP_CONFIG_PARAMS with sessionId " << sessionId << ", hr = " << std::showbase << std::hex << hr;
-    //     resultPromise.set_exception(std::make_exception_ptr(UwbException(UwbStatusGeneric::Rejected)));
-    //     return resultFuture;
-    // } else {
-    //     PLOG_DEBUG << "IOCTL_UWB_SET_APP_CONFIG_PARAMS succeeded";
-    //     auto& uwbSetAppConfigParamsStatus = *reinterpret_cast<UWB_SET_APP_CONFIG_PARAMS_STATUS*>(std::data(statusBuffer));
-    //     auto uwbStatus = UwbCxDdi::To(uwbSetAppConfigParamsStatus.status);
-    //     if (!IsUwbStatusOk(uwbStatus)) {
-    //         resultPromise.set_exception(std::make_exception_ptr(UwbException(std::move(uwbStatus))));
-    //     } else {
-    //         // TODO: Could move this logic into its own UwbCxDdi::To() function
-    //         std::vector<std::tuple<::uwb::protocol::fira::UwbApplicationConfigurationParameterType, ::uwb::protocol::fira::UwbStatus>> appConfigParamsStatusList;
-    //         for (auto i = 0; i < uwbSetAppConfigParamsStatus.appConfigParamsCount; i++)
-    //         {
-    //             auto paramType = UwbCxDdi::To(uwbSetAppConfigParamsStatus.appConfigParamsStatus[i].paramType);
-    //             auto paramStatus = UwbCxDdi::To(uwbSetAppConfigParamsStatus.appConfigParamsStatus[i].status);
-
-    //             auto appConfigParamStatus = std::make_tuple(paramType, paramStatus);
-    //             appConfigParamsStatusList.push_back(appConfigParamStatus);
-    //         }
-    //         auto result = std::make_tuple(uwbStatus, std::move(appConfigParamsStatusList));
-    //         resultPromise.set_value(std::move(result));
-    //     }
-    // }
+
+    wil::shared_hfile handleDriver;
+    auto hr = OpenDriverHandle(handleDriver, m_deviceName.c_str());
+    if (FAILED(hr)) {
+        PLOG_ERROR << "failed to obtain driver handle for " << m_deviceName << ", hr=" << std::showbase << std::hex << hr;
+        resultPromise.set_exception(std::make_exception_ptr(UwbException(UwbStatusGeneric::Rejected)));
+        return resultFuture;
+    }
+
+    auto paramsBuffer = UwbCxDdi::From(applicationConfigurationParameters).data();

```

Comment:
`data()` here returns a `std::span<uint8_t>` which is a non-owning view of the data, however, the actual object is a temporary because the result of `UwbCxDdi::From()` isn't saved here, so this will access dangling memory. This should be done in 2 stages:

```suggestion
    auto paramsDdi = UwbCxDdi::From(applicationConfigurationParameters);
    auto paramsBuffer = std::data(paramsDdi);
```